Source/Dafny/DafnyAst.cs

```diff
@@ -7932,6 +7932,18 @@ public IfStmt(IToken tok, IToken endTok, bool isBindingGuard, Expression guard,
       this.Thn = thn;
       this.Els = els;
     }
+    public IfStmt(IToken tok, IToken endTok, bool isBindingGuard, Expression guard, BlockStmt thn, Statement els, Attributes attrs)
+      : base(tok, endTok, attrs) {
+      Contract.Requires(tok != null);
+      Contract.Requires(endTok != null);
+      Contract.Requires(!isBindingGuard || (guard is ExistsExpr && ((ExistsExpr)guard).Range == null));
+      Contract.Requires(thn != null);
+      Contract.Requires(els == null || els is BlockStmt || els is IfStmt || els is SkeletonStatement);
```

Comment:
Thanks for including these contracts. Even though we're not doing run-time checking of them at this time, they clearly state the expectations on the caller.